add initial content to Documentation index

elmiko opened the pull request:
this describes the basic expectations for documentation when adding a
new component.

elmiko commented:
@jeana-redhat we talked about this a little during team standup, curious if you might have any thoughts or comments about the instruction i am proposing.

JoelSpeed reviewed (commented):
Is it worth mentioning enhancements and the role they play? I would have thought a good enhancement would also form the basis of the product docs, though maybe that comes from a different page within this project 🤔

elmiko commented:
we do have a separate doc on creating the enhancement, https://github.com/elmiko/ocp-infrastructure-provider-onboarding-guide/blob/main/docs/openshift-enhancement/index.md
do you think it's worth making a mentioning and dropping a link in this doc?

jeana-redhat commented on docs/documentation/index.md:
```diff
@@ -1,3 +1,27 @@
+
+## Repository level documentation
+
+It is expected that any component which has its code in a Git repository will have
+documentation in that repository to cover its usage and troubleshooting. This
```
If you do want to make this more of a hard requirement, perhaps:
```suggestion
Any component that has its code in a Git repository must have
documentation in that repository to cover its usage and troubleshooting. This
```

elmiko replied:
i like that, thanks

jeana-redhat reviewed (commented):
Per Slack discussion :slightly_smiling_face:

elmiko commented:
thanks for the suggestions, i updated the content to reference the enhancement (with link to doc page), and added the stronger language about repo docs.

lobziik commented on docs/documentation/index.md:
```diff
@@ -1,3 +1,29 @@
+documentation in the [OpenShift product documentation][ocp-docs]. The content
+for these product documents are created and maintained by Red Hat, but they
+are inspired by the [OpenShift enhancement](../openshift-enhancement),
+repository level documentation, and direct discussions with the component
```
Hm, I'm thinking about some examples of good repo level documentation which we like, would nice be refer something as example.

elmiko replied:
i think that's a good idea, was there an example doc that you were thinking about?

we could point to the MAO documentation, but that's quite verbose. perhaps the [alibaba readme](https://github.com/openshift/cluster-api-provider-alibaba/blob/main/README.md), that seems to have some nice info about how to deploy and use the controller. wdyt?

elmiko replied:
maybe [cloud provider ibm](https://github.com/openshift/cloud-provider-ibm/blob/master/README.md) too?

lobziik replied:
Alibaba looks nice, lets reference it there?

Our MAPI docs is also good enough example i think

elmiko replied:
sounds good, i'll add some link. thanks!

elmiko commented:
updated to add some example links

elmiko commented:
this has gone over a week with no further review. i am going to merge under lazy consensus
